src/init.cpp

```diff
@@ -862,9 +866,15 @@ bool AppInit2(boost::thread_group& threadGroup)
     boost::filesystem::path pathLockFile = GetDataDir() / ".lock";
     FILE* file = fopen(pathLockFile.string().c_str(), "a"); // empty lock file; created if it doesn't exist.
     if (file) fclose(file);
-    static boost::interprocess::file_lock lock(pathLockFile.string().c_str());
-    if (!lock.try_lock())
-        return InitError(strprintf(_("Cannot obtain a lock on data directory %s. Bitcoin Core is probably already running."), strDataDir));
+
+    try {
+        static boost::interprocess::file_lock lock(pathLockFile.string().c_str());
+        if (!lock.try_lock())
+            return InitError(strprintf(_("Cannot obtain a lock on data directory %s. Bitcoin Core is probably already running."), strDataDir));
+    } catch(const boost::interprocess::interprocess_exception& e) {
```

Comment:
OK. But the problem is that any message returned by .what() will be either in English, or completely incomprehensible. 

So may as well keep the entire message in English then. Easier to Google, too. 

The alternative would be to have a message that explain the probable source of the problem and how to resolve it in understandable language. Adding another translation for "Could not lock data directory" isn't useful.

If you mean to get this in for 0.11, where a string freeze is in progress, I'd go for the first.